From: Jim Fehlig <jfehlig@suse.com>
Date: Thu, 12 May 2016 14:40:29 -0600
Subject: [PATCH] xlconfigtests: use qemu-xen in all test data files

Some of the test configuration files in tests/xlconfigdata
use the old qemu-dm as the emulator. Many of the configuration
features tested (spice, rbd, multi-usb) are not even usable with
the old qemu. Change these files to use the new qemu-xen (also
known as qemu upstream) emulator.

Note: This change fixes xlconfigtest failures when the old
qemu is actually installed on the system. During device post
parse, the libxl driver attempts to invoke the emulator to
determine if it is the old or new qemu so it can properly set
video RAM defaults. With the old qemu installed, the default
video RAM was set differently than the expected value.
Changing all the test data files to use qemu-xen ensures
predictable results wrt default video RAM size.

Signed-off-by: Jim Fehlig <jfehlig@suse.com>
(cherry picked from commit b90c4b5f505698d600303c5b4f03f5d229b329dd)
---
 tests/xlconfigdata/test-disk-positional-parms-full.cfg    | 2 +-
 tests/xlconfigdata/test-disk-positional-parms-full.xml    | 2 +-
 tests/xlconfigdata/test-disk-positional-parms-partial.cfg | 2 +-
 tests/xlconfigdata/test-disk-positional-parms-partial.xml | 2 +-
 tests/xlconfigdata/test-fullvirt-multiusb.cfg             | 2 +-
 tests/xlconfigdata/test-fullvirt-multiusb.xml             | 2 +-
 tests/xlconfigdata/test-fullvirt-nohap.cfg                | 2 +-
 tests/xlconfigdata/test-fullvirt-nohap.xml                | 2 +-
 tests/xlconfigdata/test-new-disk.cfg                      | 2 +-
 tests/xlconfigdata/test-new-disk.xml                      | 2 +-
 tests/xlconfigdata/test-rbd-multihost-noauth.cfg          | 2 +-
 tests/xlconfigdata/test-rbd-multihost-noauth.xml          | 2 +-
 tests/xlconfigdata/test-spice-features.cfg                | 2 +-
 tests/xlconfigdata/test-spice-features.xml                | 2 +-
 tests/xlconfigdata/test-spice.cfg                         | 2 +-
 tests/xlconfigdata/test-spice.xml                         | 2 +-
 tests/xlconfigdata/test-vif-rate.cfg                      | 2 +-
 tests/xlconfigdata/test-vif-rate.xml                      | 2 +-
 18 files changed, 18 insertions(+), 18 deletions(-)

diff --git a/tests/xlconfigdata/test-disk-positional-parms-full.cfg b/tests/xlconfigdata/test-disk-positional-parms-full.cfg
index c5bbb03..217d4dc 100644
--- a/tests/xlconfigdata/test-disk-positional-parms-full.cfg
+++ b/tests/xlconfigdata/test-disk-positional-parms-full.cfg
@@ -12,7 +12,7 @@ localtime = 0
 on_poweroff = "destroy"
 on_reboot = "restart"
 on_crash = "restart"
-device_model = "/usr/lib/xen/bin/qemu-dm"
+device_model = "/usr/lib/xen/bin/qemu-system-i386"
 sdl = 0
 vnc = 1
 vncunused = 1
diff --git a/tests/xlconfigdata/test-disk-positional-parms-full.xml b/tests/xlconfigdata/test-disk-positional-parms-full.xml
index 41e8804..1bc5b43 100644
--- a/tests/xlconfigdata/test-disk-positional-parms-full.xml
+++ b/tests/xlconfigdata/test-disk-positional-parms-full.xml
@@ -19,7 +19,7 @@
   <on_reboot>restart</on_reboot>
   <on_crash>restart</on_crash>
   <devices>
-    <emulator>/usr/lib/xen/bin/qemu-dm</emulator>
+    <emulator>/usr/lib/xen/bin/qemu-system-i386</emulator>
     <disk type='block' device='disk'>
       <driver name='phy' type='raw'/>
       <source dev='/dev/HostVG/XenGuest2'/>
diff --git a/tests/xlconfigdata/test-disk-positional-parms-partial.cfg b/tests/xlconfigdata/test-disk-positional-parms-partial.cfg
index 09eeb94..fd16db0 100644
--- a/tests/xlconfigdata/test-disk-positional-parms-partial.cfg
+++ b/tests/xlconfigdata/test-disk-positional-parms-partial.cfg
@@ -12,7 +12,7 @@ localtime = 0
 on_poweroff = "destroy"
 on_reboot = "restart"
 on_crash = "restart"
-device_model = "/usr/lib/xen/bin/qemu-dm"
+device_model = "/usr/lib/xen/bin/qemu-system-i386"
 sdl = 0
 vnc = 1
 vncunused = 1
diff --git a/tests/xlconfigdata/test-disk-positional-parms-partial.xml b/tests/xlconfigdata/test-disk-positional-parms-partial.xml
index 6578e59..e86a5be 100644
--- a/tests/xlconfigdata/test-disk-positional-parms-partial.xml
+++ b/tests/xlconfigdata/test-disk-positional-parms-partial.xml
@@ -19,7 +19,7 @@
   <on_reboot>restart</on_reboot>
   <on_crash>restart</on_crash>
   <devices>
-    <emulator>/usr/lib/xen/bin/qemu-dm</emulator>
+    <emulator>/usr/lib/xen/bin/qemu-system-i386</emulator>
     <disk type='block' device='disk'>
       <driver name='phy' type='raw'/>
       <source dev='/dev/HostVG/XenGuest2'/>
diff --git a/tests/xlconfigdata/test-fullvirt-multiusb.cfg b/tests/xlconfigdata/test-fullvirt-multiusb.cfg
index 003eb2b..6d456de 100755
--- a/tests/xlconfigdata/test-fullvirt-multiusb.cfg
+++ b/tests/xlconfigdata/test-fullvirt-multiusb.cfg
@@ -12,7 +12,7 @@ localtime = 0
 on_poweroff = "destroy"
 on_reboot = "restart"
 on_crash = "restart"
-device_model = "/usr/lib/xen/bin/qemu-dm"
+device_model = "/usr/lib/xen/bin/qemu-system-i386"
 sdl = 0
 vnc = 1
 vncunused = 1
diff --git a/tests/xlconfigdata/test-fullvirt-multiusb.xml b/tests/xlconfigdata/test-fullvirt-multiusb.xml
index d7df23a..fcd14e9 100644
--- a/tests/xlconfigdata/test-fullvirt-multiusb.xml
+++ b/tests/xlconfigdata/test-fullvirt-multiusb.xml
@@ -19,7 +19,7 @@
   <on_reboot>restart</on_reboot>
   <on_crash>restart</on_crash>
   <devices>
-    <emulator>/usr/lib/xen/bin/qemu-dm</emulator>
+    <emulator>/usr/lib/xen/bin/qemu-system-i386</emulator>
     <disk type='block' device='disk'>
       <driver name='phy' type='raw'/>
       <source dev='/dev/HostVG/XenGuest2'/>
diff --git a/tests/xlconfigdata/test-fullvirt-nohap.cfg b/tests/xlconfigdata/test-fullvirt-nohap.cfg
index 44bfa3c..e7e933d 100644
--- a/tests/xlconfigdata/test-fullvirt-nohap.cfg
+++ b/tests/xlconfigdata/test-fullvirt-nohap.cfg
@@ -13,7 +13,7 @@ localtime = 0
 on_poweroff = "destroy"
 on_reboot = "restart"
 on_crash = "restart"
-device_model = "/usr/lib/xen/bin/qemu-dm"
+device_model = "/usr/lib/xen/bin/qemu-system-i386"
 sdl = 0
 vnc = 1
 vncunused = 1
diff --git a/tests/xlconfigdata/test-fullvirt-nohap.xml b/tests/xlconfigdata/test-fullvirt-nohap.xml
index 9cd7b0b..e57e28b 100644
--- a/tests/xlconfigdata/test-fullvirt-nohap.xml
+++ b/tests/xlconfigdata/test-fullvirt-nohap.xml
@@ -20,7 +20,7 @@
   <on_reboot>restart</on_reboot>
   <on_crash>restart</on_crash>
   <devices>
-    <emulator>/usr/lib/xen/bin/qemu-dm</emulator>
+    <emulator>/usr/lib/xen/bin/qemu-system-i386</emulator>
     <disk type='block' device='disk'>
       <driver name='phy' type='raw'/>
       <source dev='/dev/HostVG/XenGuest2'/>
diff --git a/tests/xlconfigdata/test-new-disk.cfg b/tests/xlconfigdata/test-new-disk.cfg
index b079056..4fe76b2 100644
--- a/tests/xlconfigdata/test-new-disk.cfg
+++ b/tests/xlconfigdata/test-new-disk.cfg
@@ -12,7 +12,7 @@ localtime = 0
 on_poweroff = "destroy"
 on_reboot = "restart"
 on_crash = "restart"
-device_model = "/usr/lib/xen/bin/qemu-dm"
+device_model = "/usr/lib/xen/bin/qemu-system-i386"
 sdl = 0
 vnc = 1
 vncunused = 1
diff --git a/tests/xlconfigdata/test-new-disk.xml b/tests/xlconfigdata/test-new-disk.xml
index 41e8804..1bc5b43 100644
--- a/tests/xlconfigdata/test-new-disk.xml
+++ b/tests/xlconfigdata/test-new-disk.xml
@@ -19,7 +19,7 @@
   <on_reboot>restart</on_reboot>
   <on_crash>restart</on_crash>
   <devices>
-    <emulator>/usr/lib/xen/bin/qemu-dm</emulator>
+    <emulator>/usr/lib/xen/bin/qemu-system-i386</emulator>
     <disk type='block' device='disk'>
       <driver name='phy' type='raw'/>
       <source dev='/dev/HostVG/XenGuest2'/>
diff --git a/tests/xlconfigdata/test-rbd-multihost-noauth.cfg b/tests/xlconfigdata/test-rbd-multihost-noauth.cfg
index 99f0889..01c15d5 100644
--- a/tests/xlconfigdata/test-rbd-multihost-noauth.cfg
+++ b/tests/xlconfigdata/test-rbd-multihost-noauth.cfg
@@ -12,7 +12,7 @@ localtime = 0
 on_poweroff = "destroy"
 on_reboot = "restart"
 on_crash = "restart"
-device_model = "/usr/lib/xen/bin/qemu-dm"
+device_model = "/usr/lib/xen/bin/qemu-system-i386"
 sdl = 0
 vnc = 1
 vncunused = 1
diff --git a/tests/xlconfigdata/test-rbd-multihost-noauth.xml b/tests/xlconfigdata/test-rbd-multihost-noauth.xml
index 728aa1e..ef9bd17 100644
--- a/tests/xlconfigdata/test-rbd-multihost-noauth.xml
+++ b/tests/xlconfigdata/test-rbd-multihost-noauth.xml
@@ -19,7 +19,7 @@
   <on_reboot>restart</on_reboot>
   <on_crash>restart</on_crash>
   <devices>
-    <emulator>/usr/lib/xen/bin/qemu-dm</emulator>
+    <emulator>/usr/lib/xen/bin/qemu-system-i386</emulator>
     <disk type='block' device='disk'>
       <driver name='phy' type='raw'/>
       <source dev='/dev/HostVG/XenGuest2'/>
diff --git a/tests/xlconfigdata/test-spice-features.cfg b/tests/xlconfigdata/test-spice-features.cfg
index 48dcd86..f8a25e4 100644
--- a/tests/xlconfigdata/test-spice-features.cfg
+++ b/tests/xlconfigdata/test-spice-features.cfg
@@ -12,7 +12,7 @@ localtime = 0
 on_poweroff = "destroy"
 on_reboot = "restart"
 on_crash = "restart"
-device_model = "/usr/lib/xen/bin/qemu-dm"
+device_model = "/usr/lib/xen/bin/qemu-system-i386"
 vif = [ "mac=00:16:3e:66:92:9c,bridge=xenbr1,script=vif-bridge,model=e1000" ]
 parallel = "none"
 serial = "none"
diff --git a/tests/xlconfigdata/test-spice-features.xml b/tests/xlconfigdata/test-spice-features.xml
index 3820732..8175760 100644
--- a/tests/xlconfigdata/test-spice-features.xml
+++ b/tests/xlconfigdata/test-spice-features.xml
@@ -19,7 +19,7 @@
   <on_reboot>restart</on_reboot>
   <on_crash>restart</on_crash>
   <devices>
-    <emulator>/usr/lib/xen/bin/qemu-dm</emulator>
+    <emulator>/usr/lib/xen/bin/qemu-system-i386</emulator>
     <disk type='block' device='disk'>
       <driver name='phy' type='raw'/>
       <source dev='/dev/HostVG/XenGuest2'/>
diff --git a/tests/xlconfigdata/test-spice.cfg b/tests/xlconfigdata/test-spice.cfg
index 7ab23e1..abdf63d 100644
--- a/tests/xlconfigdata/test-spice.cfg
+++ b/tests/xlconfigdata/test-spice.cfg
@@ -12,7 +12,7 @@ localtime = 0
 on_poweroff = "destroy"
 on_reboot = "restart"
 on_crash = "restart"
-device_model = "/usr/lib/xen/bin/qemu-dm"
+device_model = "/usr/lib/xen/bin/qemu-system-i386"
 vif = [ "mac=00:16:3e:66:92:9c,bridge=xenbr1,script=vif-bridge,model=e1000" ]
 parallel = "none"
 serial = "none"
diff --git a/tests/xlconfigdata/test-spice.xml b/tests/xlconfigdata/test-spice.xml
index f33691f..32cad27 100644
--- a/tests/xlconfigdata/test-spice.xml
+++ b/tests/xlconfigdata/test-spice.xml
@@ -19,7 +19,7 @@
   <on_reboot>restart</on_reboot>
   <on_crash>restart</on_crash>
   <devices>
-    <emulator>/usr/lib/xen/bin/qemu-dm</emulator>
+    <emulator>/usr/lib/xen/bin/qemu-system-i386</emulator>
     <disk type='block' device='disk'>
       <driver name='phy' type='raw'/>
       <source dev='/dev/HostVG/XenGuest2'/>
diff --git a/tests/xlconfigdata/test-vif-rate.cfg b/tests/xlconfigdata/test-vif-rate.cfg
index db932e5..34a19a2 100644
--- a/tests/xlconfigdata/test-vif-rate.cfg
+++ b/tests/xlconfigdata/test-vif-rate.cfg
@@ -12,7 +12,7 @@ localtime = 0
 on_poweroff = "destroy"
 on_reboot = "restart"
 on_crash = "restart"
-device_model = "/usr/lib/xen/bin/qemu-dm"
+device_model = "/usr/lib/xen/bin/qemu-system-i386"
 sdl = 0
 vnc = 1
 vncunused = 1
diff --git a/tests/xlconfigdata/test-vif-rate.xml b/tests/xlconfigdata/test-vif-rate.xml
index 3620e2a..3ab7488 100644
--- a/tests/xlconfigdata/test-vif-rate.xml
+++ b/tests/xlconfigdata/test-vif-rate.xml
@@ -19,7 +19,7 @@
   <on_reboot>restart</on_reboot>
   <on_crash>restart</on_crash>
   <devices>
-    <emulator>/usr/lib/xen/bin/qemu-dm</emulator>
+    <emulator>/usr/lib/xen/bin/qemu-system-i386</emulator>
     <disk type='block' device='disk'>
       <driver name='phy' type='raw'/>
       <source dev='/dev/HostVG/XenGuest2'/>
